### DISCUSSION TITLE
mogrify conditional cropping based on dimensions

### DISCUSSION THREAD

**Discussion by lightroomstatistics** [ImageMagick/ImageMagick#5674] [Help]
I have thousands of images exported by Lightroom.
These images where shot by both Canon and Nikon (D800) camera's.

After the export the Canon images are exactly 248 x 372, but the Nikon images are either 247 x 372 or 248 x 374.

**Yet I need them to be all exact 248 x 372.**

I used `magick mogrify -gravity Center -crop 248x372+0+0 *.jpg` and that works fine, however there is a performance issue, because all the images are processed. 

## Question
How can I use the `magick mogrify -gravity Center -crop 248x372+0+0 *.jpg` only on images that have a different dimension than 248 x 372. Something like a combination of `identify` and `mogrify`.

The solution should both work on the Windows and the MacOs command line.

**Comment by fmw42** [comment]
Mogrify is old even when run from IM 7. So there is no way that I know to do that inside mogrify.  You will likely need to write a script FOR loop over each image in your folder and test the dimensions before trying to crop.
